### PR TITLE
client cert doesn't need its own port

### PR DESCRIPTION
**Pull request by binary1230**
OK, this is actually a way nicer solution.

Per @EliAndrewC's feedback I have been able to rework this to not need a separate port.

Anyone can connect to port 443 on nginx, and nginx will, if present, attempt to validate any client cert that is presented against our CA cert.  

However, it will deny access to /jsonrpc if you have not presented a valid client cert.

So, if you have a client cert and can reach port 443, you're good to make JSON-RPC requests.

### CODE REVIEW

**Comment by binary1230** [comment]
requires the following PRs too:
https://github.com/magfest/ubersystem-deploy/pull/42
https://github.com/magfest/production-config/pull/55

File: files/generate-CA-and-clientcerts.sh [hunk 1]
```diff
@@ -26,29 +26,19 @@ subj_common_name=$6
...
 subj_ca="$subj=$subj_common_name"
-subj_server="$subj=$subj_common_name-server"
 subj_client="$subj=$subj_common_name-client"
 
 ssl_ca_key=${prefix}-ca.key
 ssl_ca_crt=${prefix}-ca.crt
-ssl_server_key=${prefix}-server.key
-ssl_server_csr=${prefix}-server.csr
-ssl_server_crt=${prefix}-server.crt
 ssl_client_key=${prefix}-client.key
```
Comment (binary1230) [inline files/generate-CA-and-clientcerts.sh]: we no longer generate the self-signed server key/cert

File: manifests/json_rpc.pp [hunk 2]
```diff
@@ -1,56 +0,0 @@
-# exposes the JSON RPC API with nginx
```
Comment (binary1230) [inline manifests/json_rpc.pp]: everything in here is gone, re-created in a better way back in nginx.pp

**Comment by kitsuta** [comment]
This looks good as far as I can tell, though it's still a bit over my head.

**Comment by EliAndrewC** [comment]
+1

**Pull request merged**